pandas/core/frame.py

```diff
@@ -6675,6 +6678,9 @@ def _dispatch_frame_op(self, right, func: Callable, axis: Optional[int] = None):
 
             right = right._values
 
+            if isinstance(right, TimedeltaArray):
+                right = right._data
```

Comment:
1) this is AM-only right?
2) wont the array_op call just re-wrap this?
3) if not, will something like DataFrame[int] + Series[timedelta64ns] fail to raise because of this?

Reply:
4) if this is a hack as the commit message suggests, then a comment would be worthwhile

Reply:
See https://github.com/pandas-dev/pandas/pull/40482#issuecomment-811177005 for the general explanation of in which case this happens. It was not my intention to keep this, I am hoping to find a better solution.

Reply:
Yah I saw that, thought about commenting, but decided that "last time i handled that by reshaping to 2D" wouldn't be something you'd find that helpful.